tests/phpunit/tests/load/wpGetDevelopmentMode.php

```diff
@@ -43,4 +44,118 @@ public function test_wp_get_development_mode_filter_invalid_value() {
 		$_wp_tests_development_mode = 'invalid';
 		$this->assertSame( '', wp_get_development_mode() );
 	}
+
+	/**
+	 * Tests that `wp_in_development_mode()` returns expected results.
+	 *
+	 * @ticket 57487
+	 * @dataProvider data_wp_in_development_mode
+	 */
+	public function test_wp_in_development_mode( $current, $given, $expected ) {
+		global $_wp_tests_development_mode;
+
+		$_wp_tests_development_mode = $current;
+
+		if ( $expected ) {
+			$this->assertTrue( wp_in_development_mode( $given ) );
+		} else {
+			$this->assertFalse( wp_in_development_mode( $given ) );
+		}
+	}
+
+	/**
+	 * Data provider that returns test scenarios for the `test_wp_in_development_mode()` method.
+	 *
+	 * @return array Test scenarios, each one with 3 entries for currently set development mode (string), given
+	 *               development mode (string), and expected result (boolean).
+	 */
+	public function data_wp_in_development_mode() {
+		return array(
+			array(
+				'core',
+				'core',
+				true,
+			),
+			array(
+				'plugin',
+				'plugin',
+				true,
+			),
+			array(
+				'theme',
+				'theme',
+				true,
+			),
+			array(
+				'core',
+				'plugin',
+				false,
+			),
+			array(
+				'core',
+				'theme',
+				false,
+			),
+			array(
+				'plugin',
+				'core',
+				false,
+			),
+			array(
+				'plugin',
+				'theme',
+				false,
+			),
+			array(
+				'theme',
+				'core',
+				false,
+			),
+			array(
+				'theme',
+				'plugin',
+				false,
+			),
+			array(
+				'all',
+				'core',
+				true,
+			),
+			array(
+				'all',
+				'plugin',
+				true,
+			),
+			array(
+				'all',
+				'theme',
+				true,
+			),
+			array(
+				'all',
+				'all',
+				true,
+			),
+			array(
+				'all',
+				'random',
+				true,
+			),
+			array(
+				'invalid',
+				'core',
+				false,
+			),
+			array(
+				'invalid',
+				'plugin',
+				false,
+			),
+			array(
+				'invalid',
+				'theme',
+				false,
+			),
```

Comment:
This makes it easier to determine which item is failing.

Note: This will fail the phpcs sniffs but that can be an exercise for later as getting that right in the GH interface is very painful.

```suggestion
			'core mode, testing for core' => array(
				'core',
				'core',
				true,
			),
			'plugin mode, testing for plugin' => array(
				'plugin',
				'plugin',
				true,
			),
			'theme mode, testing for theme' => array(
				'theme',
				'theme',
				true,
			),
			'core mode, testing for plugin' => array(
				'core',
				'plugin',
				false,
			),
			'core mode, testing for theme' => array(
				'core',
				'theme',
				false,
			),
			'plugin mode, testing for core' => array(
				'plugin',
				'core',
				false,
			),
			'plugin mode, testing for theme' => array(
				'plugin',
				'theme',
				false,
			),
			'theme mode, testing for core' => array(
				'theme',
				'core',
				false,
			),
			'theme mode, testing for plugin' => array(
				'theme',
				'plugin',
				false,
			),
			'all mode, testing for core' => array(
				'all',
				'core',
				true,
			),
			'all mode, testing for plugin' => array(
				'all',
				'plugin',
				true,
			),
			'all mode, testing for theme' => array(
				'all',
				'theme',
				true,
			),
			'all mode, testing for all' => array(
				'all',
				'all',
				true,
			),
			'all mode, testing for non-standard value' => array(
				'all',
				'random',
				true,
			),
			'invalid mode, testing for core' => array(
				'invalid',
				'core',
				false,
			),
			'invalid mode, testing for plugin' => array(
				'invalid',
				'plugin',
				false,
			),
			'invalid mode, testing for theme' => array(
				'invalid',
				'theme',
				false,
			),
```